tasks/src/lib.rs

```diff
@@ -80,22 +135,114 @@ pub trait Task: 'static {
 }
 
 /// Scheduler managing enqueued tasks and routing incoming responses to active tasks.
+#[derive(Debug)]
 pub struct Scheduler {
     flow: ClientFlow,
     waiting_tasks: TaskMap,
     active_tasks: TaskMap,
     tag_generator: TagGenerator,
+    capabilities: Vec1<Capability<'static>>,
+    server_id: Option<Vec<(IString<'static>, NString<'static>)>>,
 }
 
 impl Scheduler {
-    /// Create a new scheduler.
-    pub fn new(flow: ClientFlow) -> Self {
-        Self {
+    pub async fn try_new_flow(
```

Comment:
This function prevents users to have wrong or missing capabilities. It takes care of the following logic (as discussed with @duesee): take capabilities from the greeting, if not available → explicitly request them.